File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/service/impl/InMemoryChatHistoryService.java
package com.shterneregen.securelan.chat.service.impl;

import com.shterneregen.securelan.chat.service.ChatHistoryService;

import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;

public class InMemoryChatHistoryService implements ChatHistoryService {
    private final CopyOnWriteArrayList<String> lines = new CopyOnWriteArrayList<>();

    @Override
    public void append(String line) {
        lines.add(line);
    }

    @Override
    public List<String> getAll() {
        return List.copyOf(lines);
    }

    @Override
    public String dumpAsText() {
        return String.join(System.lineSeparator(), lines);
    }

    @Override
    public void clear() {
        lines.clear();
    }
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/service/impl/InMemoryChatHistoryService.kt
package com.shterneregen.securelan.chat.service.impl

import com.shterneregen.securelan.chat.service.ChatHistoryService
import java.util.concurrent.CopyOnWriteArrayList

class InMemoryChatHistoryService : ChatHistoryService {
    private val lines = CopyOnWriteArrayList<String>()

    override fun append(line: String) {
        lines.add(line)
    }

    override fun getAll(): List<String> = lines.toList()

    override fun dumpAsText(): String = lines.joinToString(System.lineSeparator())

    override fun clear() {
        lines.clear()
    }
}
